gdrive/analytics_api.py

```diff
@@ -0,0 +1,166 @@
+        if list_response is not None:
+            log.info("-------------------------------")
+            for act in list_response.accounts:
+                log.info("Name:\t\t%s" % (act.name))
+                log.info("Display name:\t%s" % (act.display_name))
+                log.info("-------------------------------")
+        else:
+            log.warn(
+                "List response was none. Ensure credentials are set correctly"
+                + " and you have access to the cloud property."
+            )
+    except Exception as e:
+        log.error(e.args)
+
+
+def export(
+    df: pd.DataFrame, date_of_report: datetime, end_date: datetime = None
+) -> str:
+    """
+    Transform the downloaded response from the google analytics API into a
+    Google Sheets Object.
+
+    This function first touches a Google Sheets object with the drive API, then
+    writes the analytics data to that object. As of right now there is no way to do
+    this in one API transaction.
+
+    Args:
+        df (pandas.DataFrame): Tabular data to export to Google Sheets object
+        date_of_report (datetime): Date the report was run
+    Returns:
+        str: Google Sheets ID of the new Sheets object
+    """
+    filename_str = generate_filename(date_of_report, end_date)
+    analytics_folder_id = drive_client.create_folder(
+        "Google Analytics", parent_id=settings.ANALYTICS_ROOT
+    )
+
+    # We have to do this in multiple steps with more than one client because the Sheets API
+    # doesnt support opening a file in a given directory.
+    sheets_id = drive_client.create_empty_spreadsheet(filename_str, analytics_folder_id)
+    log.info("Uploading to folder %s (%s)" % ("Google Analytics", analytics_folder_id))
+    result = sheets_client.export_df_to_gdrive_speadsheet(df, sheets_id)
+    log.info(
+        "Successfully created %s (%s)" % (filename_str, result.get("spreadsheetId"))
+    )
+    return sheets_id
+
+
+def analytics_export_post_processing(df: pd.DataFrame, sheets_id: str):
+    """
+    Add new pages and pivot tables.
+
+    This function is fairly naive and inefficient. If we ever want to make Google Sheets
+    more often than once a day, we should refactor this to limit the number of API transactions.
+
+    Args:
+        df (pandas.DataFrame): Tabular data in the spreadsheet
+        sheets_id (str): Google Sheets object ID
+    """
+
+    page1 = "Rekrewt Pivot Table - First Visit"
+    page2 = "Rekrewt Pivot Table - Sessions"
+    page3 = "GSA Use Pivot Table"
+    page4 = "Completions"
+
+    new_sheet_name_to_id = sheets_client.add_new_pages(
+        [page1, page2, page3, page4], sheets_id
+    )
+    log.info("Added %s pages to %s" % (len(new_sheet_name_to_id.keys()), sheets_id))
+    sheets_client.create_pivot_tables(
+        df, (page1, page2, page3, page4), new_sheet_name_to_id, sheets_id
+    )
+
+
+def generate_filename(date: datetime, end_date: datetime = None):
```

Comment:
Is there a reason for this format?

Reply:
Yes, it is the format that is expected from end users. I dont have any reason why it cant be changed, Id just want to check with the stakeholders to make sure there isnt any automation depending on this format.

Reply:
No other notes here, but it is what they're used to and I see no need to change it